Spatial.gd

```diff
@@ -287,25 +288,39 @@ func reset_minimap_masks():
 		minimap_path.material.set_shader_param("minimap_corner", compass_corner1)
 		minimap_path.material.set_shader_param("minimap_corner2", compass_corner2)
 
-var markerdata = {}
+var markerdata = Waypoint.Waypoint.new()
 var marker_file_path = ""
-func load_taco_markers(marker_json_file):
-	self.marker_file_path = marker_json_file
+
+func load_waypoint_markers(map_id):
+	self.marker_file_path = "res://xml_converter/protobins/" + String(map_id) + ".data"
+	print("Loading protobuf file from path ", self.marker_file_path)
+	self.markerdata.clear_category()
+	self.markerdata.clear_icon()
+	self.markerdata.clear_trail()
+	var file = File.new()
+	file.open(self.marker_file_path, file.READ)
+	var data = file.get_buffer(file.get_len())
+	self.markerdata.from_bytes(data)
+	if !Waypoint.PB_ERR.NO_ERRORS:
+		print("OK")
+	else:
+		print(Waypoint.PB_ERR)
 	
-	if is_xml_file(marker_json_file):
-		print("Loading XML file from path ", marker_json_file)
-		var parsed_taco_tuple = taco_parser.parse_taco_xml(marker_json_file)
-		var json_payload = parsed_taco_tuple[0]
-		var error_message = parsed_taco_tuple[1]
-		if error_message != "":
-			print("XML parsing failed with error message: ", error_message)
-		self.markerdata = JSON.parse(json_payload).result
+	relative_textures_to_absolute_textures(marker_file_path.get_base_dir())
+
+	gen_map_markers()
+
+func load_taco_markers(marker_file):
```

Comment:
This was left over from the functionality of loading up a marker pack. The input is the only actual difference at this point. Since we are moving to loading on map ID, this can probably be removed.